integration-libs/cds/src/profiletag/services/profile-tag-lifecycle.service.spec.ts

```diff
@@ -80,26 +80,35 @@ describe('profileTagLifecycleService', () => {
   it('should be created', () => {
     expect(profileTagLifecycleService).toBeTruthy();
   });
-
-  it(`Should call the push method if the profile consent changes to true,
-  and ignore all further changes, only sending one consent changed event,`, () => {
-    let timesCalled = 0;
-    const subscription = profileTagLifecycleService
-      .consentGranted()
-      .pipe(tap(() => timesCalled++))
-      .subscribe();
-    isConsentGivenValue = false;
-    getConsentBehavior.next({ consent: 'test' });
-    getConsentBehavior.next({ consent: 'test' });
-    getConsentBehavior.next({ consent: 'test' });
-    isConsentGivenValue = true;
-    getConsentBehavior.next({ consent: 'test' });
-    getConsentBehavior.next({ consent: 'test' });
-    getConsentBehavior.next({ consent: 'test' });
-    isConsentGivenValue = true;
-    getConsentBehavior.next({ consent: 'test' });
-    subscription.unsubscribe();
-    expect(timesCalled).toEqual(1);
+  describe('Consent', () => {
+    it(`Should emit an event if the profile consent changes to true,`, () => {
+      let timesCalled = 0;
+      const subscription = profileTagLifecycleService
+        .consentChanged()
+        .pipe(
+          filter((event) => Boolean(event.data.granted)),
```

Comment:
Just curious, what's the reason for filtering here?

Reply:
Yes, good question. I think it is because the consent behavior had a default event. But now this is null... so this isn't needed. I removed it